src/class/ServerManager/ServerManager.cpp

```diff
@@ -169,11 +169,27 @@ ServerManager::ServerManager(const IConf *conf)
 	std::list< const IServerConf * >::iterator end = servers.end();
 	for (; it != end; it++)
 	{
-		this->_virtual_servers.push_back(new Server(*it));
+		Server *new_server;
+		try
+		{
+			new_server = new Server(*it);
```

Comment:
also, i think there should not be a a Server for each IServerConf
since many ServerConf can have the same sockets.
e.g
```
server {
	server_name a;
	listen 127.0.0.1:8080;
}

server {
	server_name b;
	listen 127.0.0.1:8080;
}

server {
	server_name c;
	listen 127.0.0.1:8080;
}
```

in this case we only need 1 Server for all ServerConfs